public/app/features/browse-dashboards/components/RecentlyDeletedActions.tsx

```diff
@@ -29,20 +29,31 @@ export function RecentlyDeletedActions() {
       .map(([uid]) => uid);
   }, [selectedItemsState.dashboard]);
 
+  const dashboardOrigin: { [key: string]: string } = {};
+  if (searchState.result) {
+    const originalLocations = selectedDashboards.map((selectedDashboard) => {
+      const index = searchState.result.view.fields.uid.values.findIndex((e) => e === selectedDashboard);
+      return searchState.result?.view.fields.location.values[index];
+    });
+    selectedDashboards.map((selectedDashboard, index) => {
+      dashboardOrigin[selectedDashboard] = originalLocations[index];
+    });
```

Comment:
if you're ever not `return`ing from a `.map`, it's a great indicator that `.map` isn't the right method to be using 💡 

```tsx
selectedDashboards.forEach((selectedDashboard, index) => {
  if (originalLocations[index]) {
    dashboardOrigin[selectedDashboard] = originalLocations[index];
  }
})
```

Reply:
I get the forEach() but if the if-statement is supposed to help with the TS error it doesn't work.
<img width="776" alt="Screenshot 2024-08-19 at 12 50 53" src="https://github.com/user-attachments/assets/9858b60e-917d-49f5-abf4-9dd0c16e3e47">

Reply:
The changes discussed above solved this issue as well.